src/lxc/start.c

```diff
@@ -344,12 +344,36 @@ static int signal_handler(int fd, uint32_t events, void *data,
 int lxc_set_state(const char *name, struct lxc_handler *handler,
 		  lxc_state_t state)
 {
+	int state_socket;
 	ssize_t ret;
 	struct lxc_list *cur, *next;
 	struct state_client *client;
 	struct lxc_msg msg = {.type = lxc_msg_state, .value = state};
 
+	if (state != STARTING) {
```

Comment:
We need to ingore the `STARTING` state here. We're only interested in `RUNNING` but also want to catch cases like `STOPPING`, `STOPPED`, and `ABORTING` in case the init binary fails to exec or some other error occurs.